app_feup/lib/redux/action_creators.dart

```diff
@@ -65,19 +65,29 @@ ThunkAction<AppState> reLogin(username, password, faculty, {Completer action}) {
   };
 }
 
-ThunkAction<AppState> login(username, password, faculty, persistentSession,
+ThunkAction<AppState> login(username, password, faculties, persistentSession,
     usernameController, passwordController) {
   return (Store<AppState> store) async {
     try {
       store.dispatch(SetLoginStatusAction(RequestStatus.busy));
+
+      // TODO: multiple faculties support
       final Session session = await NetworkRouter.login(
-          username, password, faculty, persistentSession);
+          username, password, faculties[0], persistentSession);
       store.dispatch(SaveLoginDataAction(session));
       if (session.authenticated) {
         store.dispatch(SetLoginStatusAction(RequestStatus.successful));
         await loadUserInfoToState(store);
+
+        /// A little bit crude
+        /// Slide down has priority over shared preferences
+        /// if value is not null or something like that
+        if (faculties != []) {
```

Comment:
Why would faculties be empty?

Reply:
Is it to take into account a user that has just gotten this update and is still logged in? See the comment I left on  `load_info`

Reply:
It was confusion of mine, it is meaningless, thus I will remove it